lib/matplotlib/ticker.py

```diff
@@ -902,14 +902,14 @@ def __call__(self, x, pos=None):
         fx = math.log(abs(x)) / math.log(b)
         isDecade = is_close_to_int(fx)
         exponent = np.round(fx) if isDecade else np.floor(fx)
-        coeff = np.round(x / b ** exponent)
+        coeff = np.round(abs(x) / b ** exponent)
         if coeff in self.sublabel:
             if not isDecade and self.labelOnlyBase:
-                s = ''
+                return ''
             elif x > 10000:
-                s = '%1.0e' % x
+                s = '%1.0e' % abs(x)
```

Comment:
The LogFormatter is used with the SymmetricalLogLocator so it should (and probably always has) deal with negative value. But the LogLocator doesn't deal with negative values.